### PR TITLE
chore: delete e2e deploys in afterAll

### PR DESCRIPTION
**Pull request by LekoArts**
### Summary

For our e2e tests we already run an `afterAll` call for every test. There we call `.destroy()`.

I've extended the NextDeployInstance class to override the base `destroy()` method. It now deletes the deploy that the class 
created in the `setup()` method.

It does this by calling the `netlify api` with `deleteDeploy`

### Test plan

1. See if tests that should pass still pass
1. Check if deploys are actually deleted

### Relevant links (GitHub issues, Notion docs, etc.) or a picture of cute animal

Fixes https://github.com/netlify/pod-ecosystem-frameworks/issues/423

### CODE REVIEW

**Review by sarahetter — APPROVED** [review]
Thank you!

File: test/e2e/next-test-lib/next-modes/next-deploy.ts [hunk 1]
```diff
@@ -4,9 +4,20 @@ import fs from 'fs-extra'
 import { platform } from 'os'
 import { NextInstance } from './base'
 
+type DeployRes = {
```
Comment (nickytonline) [inline test/e2e/next-test-lib/next-modes/next-deploy.ts:7]: [dust] Rename this with the full name.

```suggestion
type DeployResponse = {
```

File: test/e2e/next-test-lib/next-modes/next-deploy.ts [hunk 2]
```diff
@@ -28,22 +39,23 @@ export class NextDeployInstance extends NextInstance {
...
-      })
+      throw new Error(`You need to have netlify-cli installed.
+      
+      You can do this by running: "npm install -g netlify-cli@latest" or "yarn global add netlify-cli@latest"`)
     }
 
-    const NETLIFY_SITE_ID = process.env.NETLIFY_SITE_ID || '1d5a5c76-d445-4ae5-b694-b0d3f2e2c395'
     console.log(`Deploys site for test: ${testName}`)
+
+    this._netlifySiteId = process.env.NETLIFY_SITE_ID || '1d5a5c76-d445-4ae5-b694-b0d3f2e2c395'
```
Comment (nickytonline) [inline test/e2e/next-test-lib/next-modes/next-deploy.ts:54]: What's the default site ID `'1d5a5c76-d445-4ae5-b694-b0d3f2e2c395'`?

File: test/e2e/next-test-lib/next-modes/next-deploy.ts [hunk 3]
```diff
@@ -89,6 +102,37 @@ export class NextDeployInstance extends NextInstance {
     // no-op as the deployment is created during setup()
   }
 
+  public async destroy(): Promise<void> {
+    if (this.isDestroyed) {
+      throw new Error(`next instance already destroyed`)
```
Comment (nickytonline) [inline test/e2e/next-test-lib/next-modes/next-deploy.ts:107]: Consider a more verbose error.

```suggestion
      throw new Error(`Next.js deploy instance already destroyed`)
```

File: test/e2e/next-test-lib/next-modes/next-deploy.ts [hunk 4]
```diff
@@ -89,6 +102,37 @@ export class NextDeployInstance extends NextInstance {
...
+
+    // During setup() the test site is deployed to Netlify
+    // Once testing is complete, we should delete the deploy again
+
+    if (!process.env.NEXT_TEST_SKIP_CLEANUP) {
+      console.log(`Deleting project with deploy_id ${this._deployId}`)
+
+      const deleteRes = await execa('ntl', ['api', 'deleteDeploy', '--data', `{ "deploy_id": "${this._deployId}" }`])
+  
+      if (deleteRes.exitCode !== 0) {
```
Comment (nickytonline) [inline test/e2e/next-test-lib/next-modes/next-deploy.ts:118]: [dust] Rename to `deleteResponse`.

```suggestion
      const deleteResponse = await execa('ntl', ['api', 'deleteDeploy', '--data', `{ "deploy_id": "${this._deployId}" }`])
  
      if (deleteResponse.exitCode !== 0) {
```

**Review by nickytonline — COMMENTED** [review]
This looks great! Left some comments and suggestions.

File: test/e2e/next-test-lib/next-modes/next-deploy.ts (hunk 2 above)
Comment (LekoArts) [reply test/e2e/next-test-lib/next-modes/next-deploy.ts:54]: Who knows... 😆 That code was already there, I only moved it

**Review by nickytonline — APPROVED** [review]
Thanks for this @LekoArts! 🚢

**Pull request merged**